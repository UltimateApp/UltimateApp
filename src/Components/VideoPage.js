import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

import theme from '../styles/theme.style';
import VimeoVideo from './shared/VimeoVideo';

const VideoPage = (props) => {
  const video = props.route.params.video;

  return (
    <View style={styles.contentWrapper}>
      <View style={styles.videoAlone}>
        <VimeoVideo vimeoId={video.video} sounds shouldPlay />
      </View>
      <View style={styles.title}>
        <Text style={styles.text}>{video.title}</Text>
        <View style={styles.authorWrapper}>
          <Text style={styles.textAuthor}>{video.text}</Text>
        </View>
      </View>
      <View>
        <Text style={styles.description}>{video.description}</Text>
      </View>
    </View>
  );
};

export default VideoPage;

const styles = StyleSheet.create({
  contentWrapper: { flex: 1, backgroundColor: theme.COLOR_PRIMARY_LIGHT },
  videoAlone: { height: 250 },
  textAuthor: {
    flex: 1,
    color: theme.COLOR_SECONDARY,
    fontSize: theme.FONT_SIZE_SMALL,
  },
  title: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 10,
    textAlign: 'center',
    paddingLeft: 5,
    paddingRight: 10,
  },
  text: {
    color: theme.COLOR_PRIMARY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: 'bold',
  },
  description: {
    color: theme.COLOR_PRIMARY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    paddingHorizontal: 10,
  },
});
