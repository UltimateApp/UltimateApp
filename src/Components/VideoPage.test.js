import React from 'react';
import { render } from '@testing-library/react-native';

import VideoPage from './VideoPage';

describe('<VideoPage />', () => {
  it('renders correctly', () => {
    const route = {
      params: {
        video: 1,
      },
    };

    const { toJSON } = render(<VideoPage route={route} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
