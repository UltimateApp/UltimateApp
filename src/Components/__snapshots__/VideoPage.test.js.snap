// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 250,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#A6A6A6"
          hidesWhenStopped={true}
          size="large"
        />
        <Text>
          Loading...
        </Text>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 250,
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <ViewManagerAdapter_ExpoVideoView
          proxiedProperties={
            Object {
              "onError": [Function],
              "onFullscreenUpdate": [Function],
              "onLoad": [Function],
              "onLoadStart": [Function],
              "onReadyForDisplay": [Function],
              "onStatusUpdate": [Function],
              "resizeMode": "mock",
              "source": undefined,
              "status": Object {},
              "useNativeControls": true,
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 5,
        "paddingRight": 10,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#A6A6A6",
            "flex": 1,
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
          "paddingHorizontal": 10,
        }
      }
    />
  </View>
</View>
`;
